### PR TITLE
chore: add lab project for CSS grid in frontend cert

### LINKED ISSUE #404: chore: add lab project for CSS grid in frontend cert
The CSS Grid module will a lab for campers to do.
Labs are defined as short projects campers build on their own. It will consist of a blank editor and user stories.
Here is an example for the format:
https://www.freecodecamp.org/learn/the-odin-project/top-build-a-recipe-project/top-build-a-recipe-project

The previous workshop will have been the magazine project here
https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-css-grid-by-building-a-magazine/step-80

So you will need to create a small lab project focused on css grid.

## note about the HTML boilerplate

At this point in the cert, campers will have created a few labs and worked on a few workshops.
So we no longer need to force them to write out the HTML boilerplate each time.
So we don't need to create user stories for the HTML boilerplate anymore. 

## Note about adding new images

For adding new images, make sure they are free to use. You will also need to be add that image to our CDN repo here
https://github.com/freeCodeCamp/cdn

Once the new image has been added to the CDN repo, then you will need to reference that cdn link in your project.

If you are not sure about if an image can be used, reach out on staff chat. 

## Acceptance criteria

- [ ] create a new folder of the project in the `frontend-cert/css-projects` directory
- [ ] add your completed project in an `index.html` file and `styles.css` file in that new folder
- [ ] create a markdown file for your proposed user stories. You should call it `user-stories.md` and place it in your new project folder

### PR DESCRIPTION
**Pull request by larymak**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #404

screenshot-127 0 0 1_5500-2024 07 19-10_33_59

### CODE REVIEW

File: frontend-cert/css-projects/css-grid/index.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        <div class="small-article1">
+            <h3>Article 1 Title</h3>
+            <p>Lorem ipsum dolor sit amet.</p>
+        </div>
+        <div class="small-article2">
+            <h3>Article 2 Title</h3>
+            <p>Lorem ipsum dolor sit amet.</p>
+        </div>
+        <div class="cover-image">
+            <img src="https://via.placeholder.com/300x400" alt="Cover Image">
```
Comment (jdwilkin4) [inline frontend-cert/css-projects/css-grid/index.html:27]: don't forget to add this placeholder image to our CDN repo.
Then we can replace this link with the CDN link instead

**Review by moT01 — COMMENTED** [review]
This will be shown to campers as an example project that passes all the user stories. Do you think you could come up with some fun content to put in this instead of placeholder stuff @larymak? GPT can be really helpful with this. Maybe "Cats Monthly" or something.

File: frontend-cert/css-projects/css-grid/index.html (hunk 1 above)
Comment (larymak) [reply frontend-cert/css-projects/css-grid/index.html:27]: Okay, the image has been added to the CDN repo

**Comment by larymak** [comment]
> This will be shown to campers as an example project that fulfills all the user stories. Do you think you could come up with some fun content to put in this instead of placeholder stuff @larymak? GPT can be really helpful with this. Maybe "Cats Monthly" or something.

Done

File: frontend-cert/css-projects/css-grid/user-stories.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+1. The `.magazine-cover` should have a `grid-template-areas` property of:
+   ```
+   "title title title"
+   "feature-article feature-article cover-image"
+   "small-article1 small-article2 cover-image"
+   ```
+1. The `.magazine-cover` should have a `grid-template-columns` property divided into three parts.
+1. The `.magazine-cover` should have a `grid-template-rows` property of auto for the first row and equal heights for the remaining two rows.
+1. Add a `gap` of `10px` and a `width` of `80%` to the `.magazine-cover`.
+1. Set the background color of the `.magazine-cover` to `#fff`.
```
Comment (moT01) [inline frontend-cert/css-projects/css-grid/user-stories.md:14]: I think we could get rid of this user story. It doesn't really have anything to do with grid - users can set or not set the color if they want.

File: frontend-cert/css-projects/css-grid/user-stories.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+1. All the nested `div` elements should have a `grid-area` property that corresponds to the class name.
+1. Add a `padding` of `10px` to all the nested `div` elements classes.
+1. To all the nested `div` elements classes, add a background color of `#1B1B32`, `#f9f9f9`, `#e9e9e9` and `#ddd` respectively.
+1. The `.title` class should have white text color and center-aligned text.
+1. The `cover-image` class should have a `display` property of `flex`, a `justify-content` property of `center`, and an `align-items` property of `center`.
+1. Add a `max-width` of `100%` and a height of `auto` to the `cover-image img` element.
```
Comment (moT01) [inline frontend-cert/css-projects/css-grid/user-stories.md:20]: Can all these go?

**Review by moT01 — COMMENTED** [review]
I have some more suggestions...

Instead of using divs, we should use main and section elements to be more semantic?

I also suggested to get rid of a number of the user stories and just include the ones I think are necessary - e.g. just keep the ones that provide a little boilerplate and the ones related to grid. I don't really see the others adding much benefit - it will make the projects easier for users to follow and make it easier for us to write the tests. Let me know what you think @larymak.

**Comment by larymak** [comment]
> I have some more suggestions...
> 
> Instead of using divs, we should use main and section elements to be more semantic?
> 
> I also suggested to get rid of a number of the user stories and just include the ones I think are necessary - e.g. just keep the ones that provide a little boilerplate and the ones related to grid. I don't really see the others adding much benefit - it will make the projects easier for users to follow and make it easier for us to write the tests. Let me know what you think @larymak.

I have updated the user stories as well as the code, let me know what you think

File: frontend-cert/css-projects/css-grid/user-stories.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+1. You should have a `main` element with the class `magazine-cover`.
+1. Your `.magazine-cover` should contain a `header` element for the title and `section` elements for each article and the cover image.
+1. Set the display property of the `.magazine-cover` to `grid.`
```
Comment (Dario-DC) [inline frontend-cert/css-projects/css-grid/user-stories.md:3]:
```suggestion
1. You should set the display property of the `.magazine-cover` to `grid.`
```
Just for consistency.

File: frontend-cert/css-projects/css-grid/user-stories.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+1. The `.magazine-cover` should define a grid layout with the following structure:
+   - The title spanning the top row.
+   - Two feature articles and a cover image in the second row.
+   - Two small articles and the cover image in the third row.
+1. The `.magazine-cover` should have a `grid-template-columns` property divided into three parts.
+1. The `.magazine-cover` should have a `grid-template-rows` property of auto for the first row and equal heights for the remaining two rows.
```
Comment (Dario-DC) [inline frontend-cert/css-projects/css-grid/user-stories.md:9]:
```suggestion
1. The `.magazine-cover` should have a `grid-template-rows` property. It should be set to `auto` for the first row, and divide equally the space occupied by the remaining two rows.
```

File: frontend-cert/css-projects/css-grid/user-stories.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+1. Add a gap between grid items and set the width of the .magazine-cover to occupy most of the viewport width.
```
Comment (Dario-DC) [inline frontend-cert/css-projects/css-grid/user-stories.md:10]:
```suggestion
1. You should add a gap between grid items and set the width of the .magazine-cover to occupy most of the viewport width.
```

File: frontend-cert/css-projects/css-grid/user-stories.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+1. You should have a `main` element with the class `magazine-cover`.
+1. Your `.magazine-cover` should contain a `header` element for the title and `section` elements for each article and the cover image.
+1. Set the display property of the `.magazine-cover` to `grid.`
+1. The `.magazine-cover` should define a grid layout with the following structure:
+   - The title spanning the top row.
+   - Two feature articles and a cover image in the second row.
+   - Two small articles and the cover image in the third row.
+1. The `.magazine-cover` should have a `grid-template-columns` property divided into three parts.
```
Comment (Dario-DC) [inline frontend-cert/css-projects/css-grid/user-stories.md:8]:
```suggestion
1. The `.magazine-cover` should have a `grid-template-columns` property that divides the space into three equal parts.
```

**Review by Dario-DC — COMMENTED** [review]
I left a few comments. Once they are resolved I think this can be merged :+1:

**Review by moT01 — APPROVED** [review]

**Pull request merged**